### PR TITLE
screenshots on failure & throw more exceptions

### PR DESCRIPTION
**Pull request by gregwebs**

### CODE REVIEW

File: lib/ghostbuster.coffee [hunk 1]
```diff
@@ -59,11 +59,11 @@ class Test
         test.seenCallbacks.push getCallback # traversing links causes this to get re-fired.
         switch status
           when 'success'
-            if test.runner.useScreenshots()
```
Comment (joshbuddy) [inline lib/ghostbuster.coffee:62]: Wouldn't removing this cause no screenshots to be taken for successful attempts? Did tests pass when you did this?

Comment (gregwebs) [inline lib/ghostbuster.coffee:62]: yes, I want no screenshots on pass. I just want a screenshot on failure (maybe should be configurable though). I did run rake test, and the output looked the same as previous. I am a bit confused though as it prints FAILED and PASSED.

```

Running tests...
For test_ghost.coffee
  ✓ Simple index
  ✓ Simple slow index
  ✓ Form input
  ✓ Link traversal

For test_ghostmore.coffee
  ✓ Simple form
  ◐ Form should do more things
  ✓ Simple form with wait
  ✓ Slow form
  ✓ Simple form with specified max test duration
  ✓ Before block var

For test_injs.js
  ✓ Test for 3 li's

10 success, 0 failure, 1 pending
Copying screenshots ✓
Stopping server ✓
Cleaning up temporary screenshots ✓
FAILED
non_working_ghost ... /home/gweber/.rvm/gems/ruby-1.9.2-p290/gems/psych-1.2.1/lib/psych.rb:93: warning: already initialized constant VERSION
PASSED
```

**Comment by gregwebs** [comment]
It seems that we are supposed to return false rather than throw an exception

**Comment by gregwebs** [comment]
opening new pull request with just the screenshots change

**Pull request closed without merging**